apollo-ios-pagination/Sources/ApolloPagination/AsyncGraphQLQueryPagerCoordinator.swift

```diff
@@ -74,16 +71,14 @@ actor AsyncGraphQLQueryPagerCoordinator<InitialQuery: GraphQLQuery, PaginatedQue
   ///   - nextPageResolver: The resolver that can derive the query for loading more. This can be a different query than the `initialQuery`.
   ///   - onError: The callback when there is an error.
   init<P: PaginationInfo>(
-    client: any ApolloClientProtocol,
+    client: ApolloClient,
     initialQuery: InitialQuery,
-    watcherDispatchQueue: DispatchQueue = .main,
     extractPageInfo: @escaping (PageExtractionData<InitialQuery, PaginatedQuery, PaginationOutput<InitialQuery, PaginatedQuery>?>) -> P,
     pageResolver: ((P, PaginationDirection) -> PaginatedQuery?)?
   ) {
     self.client = client
     self.initialQuery = initialQuery
     self.extractPageInfo = extractPageInfo
-    self.watcherCallbackQueue = watcherDispatchQueue
```

Comment:
For my understanding: Is this now inherited, or is the onus on the caller to decide what actor/thread/queue they want to receive data on?

Reply:
Because everything is being done using async `Task`s instead of GCD, we don't have control over the thread anymore. If you want something to be done on the main thread, you should be able to pass your closure like this:

`pager.receive(on: DispatchQueue.main).sink { ... }`